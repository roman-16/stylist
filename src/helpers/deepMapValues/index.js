import { mapValues, isPlainObject } from 'lodash-es';

const deepMapValues = (object, callback, { scope = '' } = {}) =>
  mapValues(object, (value, key) => {
    if (isPlainObject(value)) {
      return deepMapValues(value, callback, {
        scope: scope ? `${scope}.${key}` : key,
      });
    }

    return callback(value, key, { scope });
  });

export default deepMapValues;
