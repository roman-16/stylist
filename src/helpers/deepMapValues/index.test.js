import deepMapValues from '.';

describe('deepMapValues', () => {
  it('deep maps an object', () => {
    deepMapValues(
      {
        a: 1,
        b: {
          c: 2,
          d: {
            e: 3,
          },
        },
      },
      (...args) => expect(args).toMatchSnapshot(),
    );
  });
});
