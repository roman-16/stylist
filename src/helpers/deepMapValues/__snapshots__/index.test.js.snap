// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deepMapValues deep maps an object 1`] = `
Array [
  1,
  "a",
  Object {
    "scope": "",
  },
]
`;

exports[`deepMapValues deep maps an object 2`] = `
Array [
  2,
  "c",
  Object {
    "scope": "b",
  },
]
`;

exports[`deepMapValues deep maps an object 3`] = `
Array [
  3,
  "e",
  Object {
    "scope": "b.d",
  },
]
`;
